Mlem/Extensions/View Modifiers/View+DestructiveConfirmation.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  View+DestructiveConfirmation.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2023-09-06.
+//
+
+import Foundation
+import SwiftUI
+
+struct DestructiveConfirmation: ViewModifier {
+    let confirmationMenuFunction: StandardMenuFunction?
+    @Binding var isPresentingConfirmDestructive: Bool
+    
+    func body(content: Content) -> some View {
+        content
+            .confirmationDialog("Destructive Action Confirmation", isPresented: $isPresentingConfirmDestructive) {
+                if let destructiveCallback = confirmationMenuFunction?.callback {
+                    Button("Yes", role: .destructive) {
+                        Task {
+                            destructiveCallback()
+                        }
+                    }
+                }
+            } message: {
+                if let destructivePrompt = confirmationMenuFunction?.destructiveActionPrompt {
+                    Text(destructivePrompt)
+                }
+            }
+    }
+}
+
+extension View {
+    /// View modifier to attach a destructive action confirmation.
+    ///
+    /// To use, add the following to the view in which you use it:
```

Comment:
Just curious but will the compiler tell you that you are missing these things if you use this view modifier without adding these members?